source/tests/unit/ni_fake_non_ivi_service_tests.cpp

```diff
@@ -1695,7 +1693,7 @@ TEST_F(NiFakeNonIviServiceTests, InitWithReturnedSession_AccessSession_ReturnsIn
   EXPECT_EQ(SESSION_HANDLE, accessed_handle);
 }
 
-TEST_F(NiFakeNonIviServiceTests, InitWithReturnedSessionFailsInit_AccessSession_SessionIsNotInMapAndStatusValueIsReturned)
+TEST_F(NiFakeNonIviServiceTests, InitWithReturnedSessionFailsInit_AccessSession_SessionIsNotInMap)
```

Comment:
Note to reviewers: Some of these unit tests are incapable of knowing the status being returned because 1) the concrete google class for the server context defines its methods as non-virtual which we cannot Mock, and 2) that class does not expose a way to check what has been added. 

I chose to remove that portion of a few unit tests because that functionality is thoroughly tested in the system tests.

Reply:
Aw bummer. I agree with your choice though.

I think we made a mistake in writing the fake tests against the server API instead of the client API. Mostly because these tests are clumsier, especially now that we have the client wrappers. This looks like a good argument for that. For reference, there actually are a handful of fake tests written at a client level in [ni_fake_service_tests_endtoend](https://github.com/ni/grpc-device/blob/19157132294db5fcc44442712ef062a53cfd111b/source/tests/integration/ni_fake_service_tests_endtoend.cpp).

Reply:
I'm going to submit these as-is. Should this warrant a github issue being opened for it? Do those issues cover both bugs and tech debt?